### PR TITLE
Implementation of the Developer Dashboard

### PR DESCRIPTION
**Pull request by nickwxo**
This pull request implements the functionality in the new design for the developer dashboard including : searching for applications, enabling and disabling applications (including - as per @matjames007 's recommendations - a rewrite of the way this was previously implemented), small visual effects like the hover effect on each application that shows the caret drop down from which the buttons for enabling and disabling an application may be selected.

In short, this pull request covers the developer dashboard with the exception of the navigation bar which will follow shortly in a separate pull request.

### CODE REVIEW

**Comment by nickwxo** [comment]
Decided to just make the few navigation bar edits here instead since all the requisite new CSS is on this branch already.

File: public/partials/dashboard.html [hunk 1]
```diff
@@ -5,28 +5,78 @@
...
-            <div class="col-md-12">
-                                <div ng-repeat="app in apps" class="app-wrap col-md-4">
-                                    <img src="http://placehold.it/100x100" class="pull-left" style="margin-right:10px"/>
-                                    <h4>{{app.ap_app_name}}</h4>
-                                    <small><p class="text-muted">{{app.ap_app_desc}}</p></small>
-                                    <p style="word-wrap: break-word;"><span> API Key: {{app.ap_app_token}}</span></p>
-                                </div>
+            <div class="col-md-12" style="display: table;">
+
+                <div style="display: table-cell;" ng-repeat="app in apps | filter:searchText" class="col-md-6 app-wrap" ng-class="!isEven($index) ? 'col-md-offset-1' : 'dummy'">
```
Comment (matjames007) [inline public/partials/dashboard.html]: @nick22891 css has it's own stuff to achieve the functionality you want.  Try using pure css to do this.

File: public/javascript/controllers/user-ctrl.js [hunk 2]
```diff
@@ -122,6 +125,24 @@ angular.module('harvestv2')
...
+            $scope.isEven = function (num) {//for applying CSS classes appropriately in iterating through apps in ng-repeat
+                return ((num % 2) == 0);
+            }
+
+            $scope.isAppActive = function (app_status) {
+                return app_status == 'active';
+            }
+
+            $scope.setAppState = function (index, state) {//index is the index of the app in $scope.apps;
+                console.log("Nick");
```
Comment (matjames007) [inline public/javascript/controllers/user-ctrl.js]: remove @nick22891

File: public/javascript/controllers/user-ctrl.js [hunk 3]
```diff
@@ -89,18 +89,21 @@ angular.module('harvestv2')
             };
         }
     ]
-).controller("UserDashboardCtrl", ['$scope', '$location', '$routeParams', 'CurrentUserFactory', 'UserAppsFactory', 'AppFactory','PlatformFactory',
-        function($scope, $location, $routeParams, CurrentUserFactory, UserAppsFactory, AppFactory, PlatformFactory) {
+).controller("UserDashboardCtrl", ['$scope', '$location', '$routeParams', '$filter', 'CurrentUserFactory', 'UserAppsFactory', 'AppFactory','AppDisableFactory', 'PlatformFactory',
+        function($scope, $location, $routeParams, $filter, CurrentUserFactory, UserAppsFactory, AppFactory, AppDisableFactory, PlatformFactory) {
```
Comment (matjames007) [inline public/javascript/controllers/user-ctrl.js]: What is expected to be passed in as `$routeParams`?

File: public/partials/common/navigation.html [hunk 4]
```diff
@@ -13,21 +13,21 @@
 
         <div class="collapse navbar-collapse" id="bs-example-navbar-collapse-1">
             <ul class="nav navbar-nav navbar-left">
+                <li class="active">
+                    <a href="/#/dashboard">Dashboard</a>
+                </li>
                 <li>
-                    <a href="">ABIS Home Page <i class="fa fa-external-link fa-fw"></i></a>
```
Comment (matjames007) [inline public/partials/common/navigation.html:17]: @CamiMonte and @nick22891 wasn't there a requirement for us to be able to link back to the RADA home page from the application?

File: public/partials/dashboard.html [hunk 5]
```diff
@@ -5,28 +5,78 @@
...
             </div>
 
-            <div class="col-md-12">
-                                <div ng-repeat="app in apps" class="app-wrap col-md-4">
-                                    <img src="http://placehold.it/100x100" class="pull-left" style="margin-right:10px"/>
-                                    <h4>{{app.ap_app_name}}</h4>
-                                    <small><p class="text-muted">{{app.ap_app_desc}}</p></small>
-                                    <p style="word-wrap: break-word;"><span> API Key: {{app.ap_app_token}}</span></p>
-                                </div>
+            <div class="col-md-12" style="display: table;">
```
Comment (matjames007) [inline public/partials/dashboard.html]: @CamiMonte and @nick22891 remember to move inline css to the actual style.css file that is imported at runtime.

File: routes/app/app-manager.js [hunk 6]
```diff
@@ -66,6 +66,29 @@ exports.getAppByID = function(req, res, next) {
         });
 };
 
+/**
+ * Modify an app
+ */
+exports.modifyApp = function(req, res, next) {
```
Comment (matjames007) [inline routes/app/app-manager.js:76]: @nick22891 is the state the only thing that you will be modifying in an app?  Suppose I want to modify the name or description of the application?

File: routes/user/user.js [hunk 7]
```diff
@@ -31,7 +31,7 @@ exports.authenticate = function(req, res, next) {
  */
 passport.use(new TokenStrategy(
     function(token, done) {
-        App.findOne({ ap_app_token: token })
+        App.findOne({ ap_app_token: token, ap_app_status: "active" })
```
Comment (matjames007) [inline routes/user/user.js:38]: good thinking... i missed this case.

**Review by matjames007 — CHANGES_REQUESTED** [review]
Review the comments left within the code changes.

File: routes/app/app-manager.js (hunk 6 above)
Comment (matjames007) [reply routes/app/app-manager.js:76]: @nick22891 think you might have missed this one.  Is the app state the only parameter to be changed?

File: public/javascript/services.js [hunk 8]
```diff
@@ -74,6 +74,15 @@ services.factory('AppFactory', function($resource) {
 });
 
 /**
+ * Enable or disable application
+ */
+services.factory('AppDisableFactory', function($resource) {
```
Comment (matjames007) [inline public/javascript/services.js]: @nick this can do more than just disable can't it?  This route can modify the entire application parameters.

**Review by matjames007 — APPROVED** [review]
